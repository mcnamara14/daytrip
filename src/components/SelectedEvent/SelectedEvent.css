.selectedEvent {
  height: 300px;
  display: block;
  background: red;
}
