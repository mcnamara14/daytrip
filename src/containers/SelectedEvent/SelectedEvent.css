.selectedEvent {
  height: 300px;
  background: #282829;
  display: flex;
  background: url('./assets/selected-bg.jpg');
  background-size: cover;
  background-position: center;
}

.selectedImage {
  width: 340px;
  height: 225px;
  background-size: cover;
  border-radius: 3px;
  background-position: center;
}

.selectedContainer {
  width: 1200px;
  margin: 0 auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.selectedTitle {
  width: 33%;
  text-align: left;
}

.selectedTitle h3 {
  font-family: "Roboto";
  text-transform: uppercase;
  font-size: 2.2rem;
  margin-bottom: 20px;
}

.selectedTitle p {
  color: #c2c2c2;
  text-transform: uppercase;
}

.selectedInfo {
  width: 30%;
  text-align: left;
  display: flex;
  align-items: center;
}

.selectedPrice {
  font-size: 3rem;
  color: #fff;
  font-family: "Roboto";
  margin-right: 20px;
}

.selectedVenue {
  color: #fff;
}

.selectedDate {
  color: #c2c2c2;
}