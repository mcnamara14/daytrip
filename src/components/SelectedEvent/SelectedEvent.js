import React, { Component } from 'react';
import { connect } from 'react-redux';
import './SelectedEvent.css';

class SelectedEvent extends Component {

  render() {
    const { image, title, price, venue, date } = this.props.selectedEvent;
    const backgroundImage = {backgroundImage: "url(" + image + ")"};
  
    return (
      <div className="selectedEvent">
        <div className="selectedContainer">
          <div className="selectedImage" style={ backgroundImage } ></div>
          <div className="selectedTitle">
            <h3>{title}</h3>
            <p>Your selected event</p>
          </div>
          <div className="selectedInfo">
            <p className="selectedPrice">{price}</p>
            <span>
              <p className="selectedVenue">{venue}</p>
              <p className="selectedDate">{date}</p>
            </span>
          </div>
        </div>
      </div>
    );
  } 
};

export const mapStateToProps = (state) => ({
  selectedEvent: state.selectedEvent
})

export default connect(mapStateToProps)(SelectedEvent);
