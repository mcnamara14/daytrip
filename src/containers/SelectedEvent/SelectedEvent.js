import React, { Component } from 'react';
import { connect } from 'react-redux';
import './SelectedEvent.css';
import PropTypes from 'prop-types';

export class SelectedEvent extends Component {
  render() {
    const { image, title, price, venue, date } = this.props.selectedEvent;
    const backgroundImage = {backgroundImage: "url(" + image + ")"};
  
    return (
      <div className="selectedEvent">
        <div className="selectedContainer">
          <div className="selectedImage" style={ backgroundImage } ></div>
          <div className="selectedTitle">
            <h3>{title}</h3>
            <p>Your selected event</p>
          </div>
          <div className="selectedInfo">
            <p className="selectedPrice">{price}</p>
            <span>
              <p className="selectedVenue">{venue}</p>
              <p className="selectedDate">{date}</p>
            </span>
          </div>
        </div>
      </div>
    );
  } 
}

SelectedEvent.propTypes = {
  selectedEvent: PropTypes.object
};

export const mapStateToProps = (state) => ({
  selectedEvent: state.selectedEvent
});

export default connect(mapStateToProps)(SelectedEvent);
