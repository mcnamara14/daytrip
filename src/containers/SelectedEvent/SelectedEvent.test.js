import React from 'react';
import ReactDOM from 'react-dom';
import { SelectedEvent } from './SelectedEvent';
import { shallow } from 'enzyme';

describe('SelectedEvent', () => {
  it('should match the snapshot', () => {
    const mockSelectedEvent = {
      image: 'http://allswifty.jpg'
    };

    const wrapper = shallow(<SelectedEvent selectedEvent={mockSelectedEvent} />)

    expect(wrapper).toMatchSnapshot();
  });
})

