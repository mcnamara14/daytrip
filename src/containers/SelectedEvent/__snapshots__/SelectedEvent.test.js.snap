// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedEvent should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectedEvent
    selectedEvent={
        Object {
            "image": "http://allswifty.jpg",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="selectedContainer"
>
        <div
                className="selectedImage"
                style={
                        Object {
                                "backgroundImage": "url(http://allswifty.jpg)",
                              }
                }
        />
        <div
                className="selectedTitle"
        >
                <h3 />
                <p>
                        Your selected event
                </p>
        </div>
        <div
                className="selectedInfo"
        >
                <p
                        className="selectedPrice"
                />
                <span>
                        <p
                                className="selectedVenue"
                        />
                        <p
                                className="selectedDate"
                        />
                </span>
        </div>
</div>,
      "className": "selectedEvent",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="selectedImage"
            style={
                        Object {
                                    "backgroundImage": "url(http://allswifty.jpg)",
                                  }
            }
/>,
          <div
            className="selectedTitle"
>
            <h3 />
            <p>
                        Your selected event
            </p>
</div>,
          <div
            className="selectedInfo"
>
            <p
                        className="selectedPrice"
            />
            <span>
                        <p
                                    className="selectedVenue"
                        />
                        <p
                                    className="selectedDate"
                        />
            </span>
</div>,
        ],
        "className": "selectedContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "selectedImage",
            "style": Object {
              "backgroundImage": "url(http://allswifty.jpg)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3 />,
              <p>
                Your selected event
</p>,
            ],
            "className": "selectedTitle",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your selected event",
              },
              "ref": null,
              "rendered": "Your selected event",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="selectedPrice"
/>,
              <span>
                <p
                                className="selectedVenue"
                />
                <p
                                className="selectedDate"
                />
</span>,
            ],
            "className": "selectedInfo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "selectedPrice",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="selectedVenue"
/>,
                  <p
                    className="selectedDate"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "selectedVenue",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "selectedDate",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="selectedContainer"
>
          <div
                    className="selectedImage"
                    style={
                              Object {
                                        "backgroundImage": "url(http://allswifty.jpg)",
                                      }
                    }
          />
          <div
                    className="selectedTitle"
          >
                    <h3 />
                    <p>
                              Your selected event
                    </p>
          </div>
          <div
                    className="selectedInfo"
          >
                    <p
                              className="selectedPrice"
                    />
                    <span>
                              <p
                                        className="selectedVenue"
                              />
                              <p
                                        className="selectedDate"
                              />
                    </span>
          </div>
</div>,
        "className": "selectedEvent",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="selectedImage"
              style={
                            Object {
                                          "backgroundImage": "url(http://allswifty.jpg)",
                                        }
              }
/>,
            <div
              className="selectedTitle"
>
              <h3 />
              <p>
                            Your selected event
              </p>
</div>,
            <div
              className="selectedInfo"
>
              <p
                            className="selectedPrice"
              />
              <span>
                            <p
                                          className="selectedVenue"
                            />
                            <p
                                          className="selectedDate"
                            />
              </span>
</div>,
          ],
          "className": "selectedContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "selectedImage",
              "style": Object {
                "backgroundImage": "url(http://allswifty.jpg)",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3 />,
                <p>
                  Your selected event
</p>,
              ],
              "className": "selectedTitle",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Your selected event",
                },
                "ref": null,
                "rendered": "Your selected event",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="selectedPrice"
/>,
                <span>
                  <p
                                    className="selectedVenue"
                  />
                  <p
                                    className="selectedDate"
                  />
</span>,
              ],
              "className": "selectedInfo",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "selectedPrice",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="selectedVenue"
/>,
                    <p
                      className="selectedDate"
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "selectedVenue",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "selectedDate",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
